core/src/main/scala/org/bitcoins/core/protocol/script/ScriptWitness.scala

```diff
@@ -180,9 +180,7 @@ object ScriptWitness extends Factory[ScriptWitness] {
     //TODO: eventually only compressed public keys will be allowed in v0 scripts
     //https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki#restrictions-on-public-key-type
     val isPubKey = {
-      stack.nonEmpty &&
-      ((stack.head.size == 33 && (stack.head.head == 0x02 || stack.head.head == 0x03))
-        || (stack.head.size == 65 && stack.head.head == 0x04))
+      stack.nonEmpty && (stack.head.size == 33 && (stack.head.head == 0x02 || stack.head.head == 0x03))
```

Comment:
Will 65 byte pub keys now cause an error?

Reply:
It will now get classified as `P2WSHWitnessV0`

Reply:
Isn't this just switching out one error for another now?

Reply:
Yes, but this thing doesn't throw an exception that kills the process. It would technically be incorrectly typed, if there is a segwit v0 that exists on the blockchain with an uncompressed public key (discouraged by policy)

Reply:
It'd be better to have a solution that still allows for uncompressed public keys

Reply:
I think you can accomplish that by having `isPublicKey` check that when `((stack.head.size == 65 && stack.head.head == 0x04))` (uncompressed pubkey) then the stack size should be greater than 1. If it is a taproot sig, it will be the only element, a p2wpkh will have the pubkey and the sig

Reply:
Any suggestions on how to distinguish between a BIP342 schnorr sig and a uncompress public key?

Reply:
You also could run `ECPublicKey.isFullyValid`